kore/src/Kore/Substitute.hs

```diff
@@ -1,159 +1,56 @@
 {- |
-Copyright   : (c) Runtime Verification, 2019
+Copyright   : (c) Runtime Verification, 2021
 License     : NCSA
 -}
 module Kore.Substitute (
-    substitute,
+    Substitute (..),
+    NormalSubstitution,
+    NormalRenaming,
 ) where
 
-import Data.Functor.Foldable (
-    Corecursive,
-    Recursive,
- )
-import qualified Data.Functor.Foldable as Recursive
-import Data.Functor.Identity (
-    Identity (runIdentity),
+import Data.Kind (
+    Type,
  )
 import Data.Map.Strict (
     Map,
  )
-import qualified Data.Map.Strict as Map
-import Data.Set (
-    Set,
- )
-import qualified Data.Set as Set
 import Kore.Attribute.Pattern.FreeVariables (
-    HasFreeVariables (..),
+    HasFreeVariables,
  )
-import qualified Kore.Attribute.Pattern.FreeVariables as FreeVariables
-import Kore.Attribute.Synthetic
 import Kore.Internal.Variable
-import Kore.Syntax
-import Kore.Variables.Binding
-import Kore.Variables.Fresh
-import Prelude.Kore
-
-{- | Traverse the pattern from the top down and apply substitutions.
-
-The 'freeVariables' annotation is used to avoid traversing subterms that
-contain none of the targeted variables.
-
-The substitution must be normalized, i.e. no target (left-hand side) variable
-may appear in the right-hand side of any substitution, but this is not checked.
--}
-substitute ::
```

Comment:
This implementation was moved to the `TermLike` instance of `Substitute`, right?

Reply:
Yes, that's right.